### DISCUSSION TITLE
[Feature]: Rule-based smart albums (workflows)

### DISCUSSION THREAD

**Discussion by srlamin19** [immich-app/immich#1673] [Feature Request]
### Feature detail

Would be great if **_Immich_** can suggest a **_smart_** view/creation in **albums** based on **Dates** (_days_, _month_, _year_), **Locations** (_Cities_, _Countries_) in this way of smart views..

### Platform

Web

**Comment by alextran1502** [comment]
Can you elaborate on the feature? It would be nice if you also had some examples.

**Reply by Gitonthegrid** [reply]
can I also give some information on this? 

I now use primarily Synology Photos because of this feature. It is extremely handy to create albums this way. 
To give a example, Every year I want to make a book from the whole family, so I can create a "smart album" that is called **Family pictures 2023** where it looks for:

- Photos that has me as face recognition
- Photos that has my wife as face recognition
- Photos that has my kids as face recognition
- Photos that are made in <this-year>

If a photo has all these boxes checked. It will put it in that created folder.

Synology Photos also works with tags (I miss that feature to in immich), so if I want to have a smart album with specific tags like Christmas, I can create a smart album that puts all photo`s with the tag **Christmas** in it. 

**These are just very simple examples, but smart albums are one of the best features I am working with the past 1.5 years**

**Comment by palitu** [comment]
i think it is something like, immich notices that you are typically in place x, then notices that you went to place y, it then says here is a "smart album" which is autmatically generated for your trip to y on date.

**Comment by puregreen59** [comment]
Thought of a great but simiple way of auto creating albums prehaps with a (Needs moderation) label.

Put a page in user settings or admin where you can add dates for birthdays of you and significant others. Even have options for anniversarys and such

You could use this information to auto create simple albums of the day like ' my son, name here 6th birthday (Moderation required)'

It's just I often find auto creation based on content never gets it right, so having it this way could be helpful, and a lot less resource intensive

Not saying instead of the normal auto albums,just as either a option to use that or this, or as a extra.

**Comment by Andrew-71** [comment]
I think it would be a good idea to have a smart album that included photos from a specific source, for example the phone camera.

A use case I can think of and would personally use that for is:
I take a lot of photos on my phone, and generally use it as my main camera. However I also save a ton of meme images. Therefore my gallery is a mess, and whenever I need to find an image from either of these categories I have to go through over 5000 files, with over half of those being completely unrelated. 

In a nutshell, what I'm suggesting is to have an album that sorts photos based on device. For example, if I select "iPhone 12" only show pics shot on an iPhone 12.

**Comment by jrasm91** [comment]
I wonder if you actually need any album features for this, or if it really just searching and filtering. I can imagine a situation where you just search by selecting device:phone and you see that subset.

**Reply by Gitonthegrid** [reply]
With Synology photos the search option also includes albums, so it is extremely handy, because you don`t have "advanced search" every time. If I want to show all photo`s with just our family on it. I can search family, and the albums pop up and I can see al the pictures with me, my wife and my child on it. 

So is you do often searches with 2+ advanced search options, it is very handy to have **smart albums**
I do know a some other people with Synology NAS and they all use this feature.

**Comment by RegalMonkey** [comment]
I think what I'd be looking for here is the ability to create an album based on a given context. Example, I can sort based on a person, allow the creation of an album to have all photos of that person in a sharable album/link. Adding to the album once a new picture is given a relevant tag in the future could be updated as a job function. Similar behavior for any other identifying metric for the picture, date, location, camera, or even manual tags that could be added.

**Reply by AngelaDMerkel** [reply]
This is a more powerful solution and can go a long way to automating some kinds of sharing. Having conditions like:

`Person 1` _or_ `Person 2` _and_ between `December 19`  and `December 25` would allow a Christmas album to be created which only contains family and which contains all Christmas holidays between a range of dates

**Comment by sorinciupitu** [comment]
UP

**Comment by sembeek** [comment]
I really miss this feature from Synology Photos, it would be nice if this is implemented in Immich.
I used it to share an album containing photos with the faces of my family members automatically, so they can access this shared album always containing their photos only.

**Reply by sorinciupitu** [reply]
me too

**Reply by Batwam** [reply]
same, I share with family members the photos I have of them. An album get updated automatically to add any photo where this person is identified
![image](https://github.com/immich-app/immich/assets/56710655/85922e8b-630d-4e43-afec-91ba5bdf2dce)

**Reply by xyxzxyz** [reply]
@alextran1502 Will this "**Auto-updating album based on person or pet**" feature going to on the roadmap soon? I am getting desperate. lol

**Reply by alextran1502** [reply]
Yes. We started the work last year. Just didn't finish it up 😁

**Reply by benmccann** [reply]
@alextran1502 is this something that you could use help on from additional contributors? I'd be interested if there are a few pointers you can share on whether there's existing work that can be picked up or contributed to

**Reply by alextran1502** [reply]
@benmccann Thank you for the offer. We have a stale (2nd oldest) https://github.com/immich-app/immich/pull/3641 that Jason and I started the work here. I am kind of debating how to carry this forward since we are refactoring and adding enhancements for search features that potentially can be used as a rule for this.

Do you use Discord, I and the team would be interested in discussing this with you.

**Reply by stephenpapierski** [reply]
@alextran1502 I just got the app up and running yesterday and am so excited that this might finally be the app that will let me replace App-Which-Must-Not-Be-Named. This might be the only feature I feel like I'm missing, so I'd me more than willing to help out in any way I can as well. I'm on discord, let me know.

**Reply by pmdTorrents** [reply]
@sembeek 
hey i made this script that does this

https://pastebin.com/Vi5hA10r

you just need to fill in:
server_url = (https://localhost:port)
email = (immich login)
password = (immich pass)
person_id= (https://localhost:port/people/**this number***)
album_id = (https://localhost:port/albums/**this number***)

and set it to run every x amount of time using either windows task manager or crontab

**Reply by Batwam** [reply]
Great, would this script work to search for all pictures including two or more specific people (e.g grandkid and grand parents)?

**Reply by dodert** [reply]
I just create this other script that use an api key and is a little simpler, is base on @pmdTorrents script.

Base on your comment @keesfluitman , my script is just to create from person, nothing else.

https://github.com/dodert/scripts/tree/main/python/immichi

**Reply by keesfluitman** [reply]
@dodert 
I had changed the previous one. Bit more complicated but allows you to set bucket size (to scan only day or month wise), and to set to update or all. Allowing you to have it try to add all assets to an album, or to only add those of that day or that month.
https://pastebin.com/nfdfCLvZ

Depending on whether this functionality will come in immich, id love to practice some python like this, allowing the script to take more env variables for more flexibility. Like multiple persons to define.

**Reply by dodert** [reply]
I just add environment to mine :D

**Reply by shummo** [reply]
@keesfluitman
Can you tell my please, how can I use this script? What and where i have to replace in script? 

Is it ok if I create a .env file next to the .py script with theese:

SERVER_URL="https://1..........................02"
BASE_URL="https://1.............................02"
API_KEY="1C....................................................LsE"
PERSON_ID="51...................................a6"
ALBUM_ID="..........................................b"
MODE="update"
BUCKET_SIZE="MONTH" 


Thanks

**Reply by keesfluitman** [reply]
Lemme improve it a bit. What system do you run immich on?

**Reply by shummo** [reply]
Thanks. I run immich in Docker (on Debain 12).

OFF:
And i have a question to you, beacuse it is really close to this script!

I would like to choose the best pics from my children/family. And they should go to an shared album.

Tagging pics, and replace them to an album is not possible yet. But  MY SOLUTION would be to mark pics as favourite, than all in favourite would go to album. With the app, me and my wife can easily add the photos to favourites. 

**Maybe is it possible also to add photos from favourite to an UI specified shared album, and than  REMOVE theese from favourites with script?**
Unfortunatelly, I can't programing, but i think the flow would be:
1.get all assets with the favorite parameter>>> https://immich.app/docs/api/get-all-assets
2. get their IDs from that call
3. add ID to the album >>> https://immich.app/docs/api/add-assets-to-album
4. then update asset command to remove the favorite status >>> https://immich.app/docs/api/update-asset

**Reply by keesfluitman** [reply]
Ye you are right, i could use those endpoints.

I hope i have time this week. 
A script like this, you place somewhere, and then add a cron job on debian. 
You know how to do that?
Otherwise add me as friend on discord and we can chat there...ill also ask if this type of auto album features are worked on.

Inform yourself a bit about running pythonscripts and crom jobs though, that's sth you should be able to do.
'To execute a Python script, first open a terminal, then navigate to the directory where the script is located, and finally, run the script using the 'python' command followed by the script's name. On Linux, consider using python3 to ensure you're using Python 3.'

**Reply by shummo** [reply]
Unfortunatelly after v1.105., i found that there is no more api/get-all-assets. Now we should use search/metadata instead.
I tired to change the api request, but now i got this error:

`Traceback (most recent call last):
  File "/home/compose/immich-app/py/NEWfavorite.py", line 115, in <module>
    main()
  File "/home/compose/immich-app/py/NEWfavorite.py", line 107, in main
    unique_asset_ids.add(asset['ids'])
                         ~^^^^^^^
TypeError: string indices must be integers, not 'str'`


Here is the text of my .py file: [fav.txt](https://github.com/user-attachments/files/16107789/fav.txt)

Can somebody help me pls with python?

(I also found oout, that /api/album/{album_id}/assets change to /api/albumS/{album_id}/assets)

**Comment by Zakmaf** [comment]
Hi everybody!
This is the first time I ever post a suggestion for an open-source project. This is how in love i'm with this project, and very grateful for all the hard work from contributors.

Coming from a service that shall remain nameless (but it's Google Photos, really), I would love to have the option to "share a face". Use faces as if they are their own albums, just albums that automatically populate.

I use to share this way on Google Photos with trusted friends. (I know it's risky because in case of "misrecognition" you could potentially share something you should'nt with someone else.)

**Comment by Kaaybi** [comment]
I also use Synology Photos and smart albums is really a cool feature.
You can create an album based on conditions like:
- source library
- specific folder path
- media type (photo and/or video)
- creation date (before or after specific date or in a range)
- **tags**
- ratings
- locations

This is specifically convenient to automatically share a constantly growing album for example like many said an album of a child shared to grandparents where pictures of your child gets automatically added with conditions like face recognition.

For Immich, in my opinion one of the simpliest way to achieve this would be to use the search feature and having a "smart album" act as a shortcut to a search request. One could create a smart album by making a search then pressing a button "save as a smart album".

One other but harder possibility is to mimic what Synology Photos did which means a modal to create and edit smart albums _(sorry the interface is in french)_:

> Smart album creation button in Synology Photos:
> ![image](https://github.com/immich-app/immich/assets/13943086/df93db11-16a5-4bde-8b30-205094f41728)

> Smart album creation modal in Synology Photos specifying a source library, dates and a tag "Partage":
> ![Capture d’écran 2023-12-30 à 10 47 34](https://github.com/immich-app/immich/assets/13943086/05e9ef1b-06db-48d1-b38c-31d28aff4489)

> Smart album modification in Synology Photos:
> ![image](https://github.com/immich-app/immich/assets/13943086/220d5c9a-c179-44ec-9af0-6e3c1f428a71)

**Reply by benmccann** [reply]
Here's what the dialog looks like in English for reference:

![Screenshot from 2024-01-25 10-20-33](https://github.com/immich-app/immich/assets/322311/c70597d8-d19b-40e3-8d97-56a26c9c709f)

**Comment by projek01** [comment]
Hello everyone, thank you for all of the contributions to make IMMICH such a great piece of open source software. This is the last feature that would allow me to pull the plug on Google Photos. I hope we can see something like this in a future release of IMMICH soon!

**Comment by gitdoc67** [comment]
I would really like to create albums based on the folder location from my external library. So /import/Work would automatically go into a Work album.

**Reply by gbcatrinoiu** [reply]
Person-based folders would be amazing! Thanks for such amazing work!

**Reply by Vergil365** [reply]
would be nice to make a smart album for photos that are in no albums :)

**Reply by shortringbear** [reply]
YES I was just about to write this, but checking if someone else did!

For example, I have a /cats folder as external libraries, that includes several subfolders for each cat. I would love to be able to create a smart album that includes one or multiple subfolders.

**Comment by frejaya** [comment]
For dates - I would like it to be able to run the logic on just month day as well, excluding year. This would allow me to create an album that automatically adds each family member's photos to an album for their birthdays throughout the years. E.g. all photos taken any year on Month x day x will automatically be added to album "Person's birthdays".

**Comment by mattkmwu** [comment]
Would definitely love to have this feature as this was the one of the cool feature in Google Photos. An album that is auto-updating that uses facial recognition. This way, my kids' album is always updated without me manually sorting it.

**Reply by betoevo** [reply]
Big thumbs Up for the devs on this lovely peace of art (Immich) I was using this feature facial recognition plus shared albums to populate a smart album between the same person on different family members gallery, for instance a smart album populated automatically with kids photos (where they are recognized) notifying grandpas but mew updates and similarly this album also populated with grandpas recognized photos of the kids. At the end,
It is already face recognition on Immich and Albums, what is missing is that they can be populated automatically by a person for different users contributing to this album

**Comment by benmccann** [comment]
A big question about smart albums is which of these two options would people prefer:
- A readonly album which is automatically updated when photos are uploaded or updated. E.g. if you add a facial recognition tag the image is added to the album or if you remove a person from being tagged in the photo then that photo is removed
- An album where photos are added to it only when the album is created and new photos are uploaded, but not when an image is updated and photos are never removed (e.g. it would not be updated when updating the facial recognition tags per the previous example). However, you can also add additional photos that you select to the album.

**Reply by Batwam** [reply]
For me it's definitely the first one. If I fix names in facial recognition after the album has been created, I would expect the album content to update accordingly.

**Reply by stephenpapierski** [reply]
I don't think it's either. How is the first one different from just going to Explore -> People and selecting a person? That is an always up to date "album" of a given person.

What I want is the option to select people to always add to an album that is not read-only. For example, I have an album for `kid1`. I would like all photos that recognize their face to show up in that album (even after photos updates), **BUT** I also want to be able to manually add photos to that album as well where their face wasn't recognized (back to the camera, in a helmet, etc).

**Reply by benmccann** [reply]
@stephenpapierski option 1 is basically equivalent to running a saved search, but could be optimized to be faster. If you're just selecting a person, it would be equivalent to selecting just that person, but you could also combine it with other facets such as a time frame, etc. If there was someone not tagged in a photo and you wanted to them to appear in an album you would simply add the missing tag

For option 2, you could add the person to the album without adding the tag. But if someone were mistagged and you removed the tag, it would not remove them from the album. Likewise, if you added the missing tag, it would not automatically add them to the album.

**Reply by dontflipmybits** [reply]
I hope your factor in how people would and want to use the face tagging ability as currently, G Photos lets you create an album based on faces you want auto-added to it. Then I can share the folder with my spouse, she also has the ability to add faces she chooses (could be different than my rules) to auto-add to the same album. Plus we both can add photos ad-hoc to it whenever.

**Reply by dontflipmybits** [reply]
It would be smart to add typical checks such that, upon creation, you can add photos that match these rules going forward or allow you to include everything in the library.
It should also evaluate when to add photos to the album as a post-check after a photo has gone through the face tagging process, it would be kludgy if this was simply cron'd.

**Comment by RegalMonkey** [comment]
I'm inclined to have it as simple as 'album is any photos containing tag(s) X,Y, and/or Z' that update on some scheduled basis to fit the parameters. This could include not only facial recognition but any arbitrary tag you applied to a photo.

I'm some ad-hoc sense this already happens by virtue of the analytics engine. If I search 'bird' the app will present me with a collection of pictures that it at least thinks are a bird. By allowing manual tags and saving of outputs for a search as an album that handles the creation, maintenance could be a scheduled task to find new add/deletes and update accordingly.

**Comment by m3brown** [comment]
There is a similar feature request (#8331) that addresses saved searches. Commenting here to acknowledge the similarity.

**Comment by keesfluitman** [comment]
> @sembeek hey i made this script that does this
> 
> https://pastebin.com/Vi5hA10r
> 
> you just need to fill in: server_url = (https://localhost:port) email = (immich login) password = (immich pass) person_id= (https://localhost:port/people/**this number***) album_id = (https://localhost:port/albums/**this number***)
> 
> and set it to run every x amount of time using either windows task manager or crontab

Nice work. Definitely will use that for now. Even if it could be nicely implemented as well! I assume you use this documentation
https://immich.app/docs/api/

Im getting this on your script
Failed to fetch time buckets. Status code: 400, Response text: {"message":["id must be a UUID"],"error":"Bad Request","statusCode":400}

The api endpoint was deprecated. See above for an adapted script.

**Comment by janstadt** [comment]
Any updates on this feature? Would really like to be able to limit a slideshow to a specific persons face and the tools that im using require an album for slideshows.

**Reply by pokemonbattledroid** [reply]
i'd love to see an update on this too.  For me at least, a missing feature that both google and synology have, which is the one thing holding me back for how I use the platforms.

**Reply by xyxzxyz** [reply]
I doubt this will be available anytime soon. I will sponsor this project again once this feature is added. For now I need to stay with Googe Photos.

**Reply by pokemonbattledroid** [reply]
Yes, doesn't seem like it's high on their agenda.  I'm not a programmer at all so not sure how these work, but it feels like at their level of skills/craft, something like this can be achieved through a filter of sorts that isn't so difficult to implement...  Well, that's my uber naive look at it.

When it comes to viewing media, I'd assume that people want to quickly filter to view assets - as a parent, for sure this is the most useful feature - but can see how it would be useful for many - even as a portal for use as an asset library at work looking for mood images, etc.

**Reply by bo0tzz** [reply]
> doesn't seem like it's high on their agenda.
>  isn't so difficult to implement...

Assessments like this are hard to get right without an insight into the workings of a project ;) In fact, we've been hard at work with significant refactoring to make this feature possible.

**Reply by pokemonbattledroid** [reply]
That's awesome to know.  I knew what I wrote would be a twat thing to write, but at least now I know it's on the radar of the team!!  :)

**Comment by NicholasFlamy** [comment]
I would like to suggest, an option to add/stack photos based on similarity, using the same backend functionality as #8228. (Would potentially cover #2479.)

**Comment by SpiderUnderUrBed** [comment]
I would go one step futher and suggest these albums have like a name, then the AI will sort the photos into their corresponding album which has the highest confidence rate, like, a album called nature will have all the nature photos in it, or you can set context separately in its own field, this would be so useful. 
(in reference to #10678)

Something like the upcoming workflows on the road map would take care of it, with the ability to create rules and have it sort, cant wait!

**Reply by blublub** [reply]
that would be a great feature

**Comment by superdong69** [comment]
Would like to be able to create an album,  tell it to add all photos from Italy,  or one for family where it includes all the people I select,  auto updating the album when more photos are added of X person or place etc.

**Comment by NicholasFlamy** [comment]
Probably over the top, but just putting it here, it might be nice if someone could define a new album and rule that would apply to all users, and maybe the album is created independently for each user with the same name and rules, mainly for going on a trip.

The use case for this is when the family goes on a trip, one member can define the album and rules, and then it will apply for everyone, and either there is an option for it to be a shared album between everyone, or it's just an album and people can share it if they want.

**Reply by superdong69** [reply]
There's no need to complicate this. 

Create album
Add photos,, add person, add, location.

When you add one of those,  give option to update when new photos that match that person, place,  to auto add to the album. 

That's all that needs to be done.

**Reply by NicholasFlamy** [reply]
I was suggesting a potential future enhancement, which wouldn't be part of the original implementation.

**Comment by droopified** [comment]
i too wish for a "smart" albums that auto populates when new images are added. so if an ablum is created to include only x person and when new images are added to the library, and the face is added, if its x person, add these new images to the album. 

but not just people. i would like a combination of things or places too. for example. person: x, keyword: beach. album only shows person x at the beach.

**Comment by thecoffeebin** [comment]
Upvote for this if I could.

I'm looking for a similar feature like the "Share as album" under "Explore - Person" in Google photos and all my newly taken photos for that person will be automatically added to this album and available for those who have been invited to this shared album. It would be extemely handy to share my Baby photos with my family!

![image (1)](https://github.com/user-attachments/assets/2ad9ada0-952a-4ee8-8b8b-8a3eb812cb68)

**Comment by Gitonthegrid** [comment]
It would already be way way better if the search field gives suggestions on the tags people, locations etc. 

I have said it before, but Synology Photos is doing a pretty good job with the search field. Every tag, person, album etc,  will pop up immediately after you type. Is use this a lot. In Immich you have to separately do a advanced search for everything item. This is so annoying. 

For now I cant get rid of Synology photos, because of some options works better. 

_That being said. The face recognition on Immich, is on a whole other level. It detects faces extremely well, even though a person is almost not visible, or half visible._

**Comment by SpiderUnderUrBed** [comment]
+1

**Comment by Batwam** [comment]
I just had a look at the project roadmap and couldn't find this feature in the list: https://github.com/orgs/immich-app/projects/1

Is this something which the team is considering to implement in the (near) future?

**Reply by SpiderUnderUrBed** [reply]
Workflows should implement this and the more specific feature I suggested earlier

**Comment by superdong69** [comment]
Can we get any official statement on this please?

**Reply by bo0tzz** [reply]
What sort of statement are you expecting? As mentioned upthread, it's on our roadmap.

**Reply by Gitonthegrid** [reply]
In can't wait it to be there.

Is this also fixing the search option that will give suggestions when you are typing?

I really do miss this feature.

**Comment by TheCWB** [comment]
I read through this thread and didnt see it specifically mentioned, so I'd like to add on to this:

I'm in favor for a manually created Smart / Conditional Album, that can be configured with specific criteria like the synology app above. The smart album would automatically populates and refresh photos that meet said criteria, on a regular or scheduled basis. Some of the criteria already exists, but I'm going to list anyways: 

Existing criteria already found in the filters: Camera Make, Camera Model, Location, Person. Type of Photo / Video, such as 360 photos, Slow Motion. Location, person.

**Reply by betoevo** [reply]
A big point in terms of usability is that if the album is shared, other people can ALSO contribute to this album, ideal for smart persons album (eg kids). This was a big point on GPhotos that I miss a lot on Synology Photos where even if automated albums can be created and shared, invited users cannot add any content so if doing a family member unified smart albums has as much albums of this person as members with its own gallery instead of a unique member gallery shared between all the family

**Comment by Snakespeare415** [comment]
This would be excellent for sharing specific key words with my users. For example: Typing "food" in the search bar and getting the option to "save search as dyamic album".

**Reply by kvalev** [reply]
That's probably the easiest way to do it, Just save the query instead of having to implement workflows, background jobs and whatnot. This is how its implemented in photoprism as well, and it works quite nicely

**Reply by radh21301** [reply]
Even if you have a query, you need a schedule trigger to search for that query and add it to the album. You still need a workflow -> trigger (when a new image is added AND it matches that query)-> action (the add it to the album 'X')

Fyi: using search with CLIP will not work. CLIP is not binary. It does not say if an image has food or no food.. just gives the probability that an image has food. So, unless you set a cutoff threshold, this feature might not work.

**Reply by kvalev** [reply]
That's not really needed, the only data your smart album needs is the query. You dont have to explicitly assign assets to the album. When you open it, all relevant assets will be loaded based on the query. However I got the feeling that the development team will go for another solution.

**Reply by radh21301** [reply]
I am not a dev. But my guess is that running a query every time you open an album will be very inefficient. But I am sure the implementation the devs come up with will solve all our use cases.

**Comment by Kugeleis** [comment]
Another take on this:
Extend the search interface to allow making an album from the results.
The same is true for maps: I'd like to collect all phots from a particular location (more detailed than just _city_) in an album.

**Reply by clodi95** [reply]
> Another take on this: Extend the search interface to allow making an album from the results. The same is true for maps: I'd like to collect all phots from a particular location (more detailed than just _city_) in an album.

this is not a new idea (see https://github.com/immich-app/immich/discussions/1673#discussioncomment-9037853 ) but... yes please!  this is the way to go (to implement smart/rule-based albums). It would be great if we could also get a way more advanced search feature. Like search for all pictures that contain this person & this person but not this person and that are tagged with this tag and which took place within a certain range from certain coordinates and which have been taken before date X but after date Y etc.

**Comment by Knuppel1983** [comment]
Can you give a rough ETA when you expect searching for tags or creating smart albums will be implemented? I’m currently deciding whether to put all my photos in Synology Photos or Immich. If you could give a rough estimate it would really help. Then i can decide if i’m moving forward with Photos or holding off till Immich has this feature.

Photos is just about there with their tag search, the problem is they don’t read xmp subject for videos.

Note AI is great, but many users use their photo management tool just for organizing. ‘France 2020’, ‘Wedding 2024’, etc. Searching for tagged photos or creating dynamic albums would be great then. Combining face recognition with tags gets you the exact photos you want.

**Reply by Gitonthegrid** [reply]
I still can't get rid of Synology Photos (to bad) 

The face recognition on immich is so great, but the search option, smart albums and the Android app is not intuitive enough for me.

For now Synology photo is way more stable and the search feature is way better/faster then Immich. 

I keep looking every month if it gets better, but I do that for a year now, and still not the improvements I was hoping for.

So if you ask my opinion, stick with Synology Photos untill they have improved some of these features.

Like I said in some other posts, please make the search feature more intuitive that you can direct search for people, places, tags etc. The search feature is one of the core features that our people use the most.

**Reply by Knuppel1983** [reply]
I agree. But it think it’s easier for Immich to fix their search function than it is for Synology to overhaul their AI / face recognition.

**Reply by Gitonthegrid** [reply]
Yes it is, but waiting for that "basic" improvement for a year now is not helping me with making the step. I have almost not see any improvements on the search option.

So maybe it is not that simple, or it is not so important for them as it is for us here.


So I really can't make the step to go to immich, because 8 people here are really relying on the search option within a Synology Photo and didn't see much improvements after a year now.

I really do hope though that they will "fix" that. Then for 100%, Immich will be better then Synology photos

**Reply by alextran1502** [reply]
@Knuppel1983 I am hoping for early next year after we get the work with stabilizing the API finished to reach our first stable release milestone

**Reply by alextran1502** [reply]
@Gitonthegrid yeah, it is not that simple since we also want to leverage this feature work to incorporate the plug-in system as well. So there are a lot of background work need to be resolved first

**Comment by alextran1502** [comment]
I am locking this thread since we've had a plan for this and it is already on the roadmap. The thread has become a place for expressing discontent on why this feature hasn't come yet, so locking to keep our focus and effort on the technical work. 

Cheers! 😄

**Reply by danieldietzler** [reply]
For reference the item on the roadmap is called "Workflows", which is a much broader feature.
